tests/console/rpm.pm

```diff
@@ -28,63 +28,77 @@ use testapi;
-
-    # List what a package provides
-    assert_script_run 'rpm -q --provides aaa_base';
-
-    # Prepare test rpm file of installed package
-    zypper_call 'in -fy --download-only aaa_base';
-    assert_script_run 'mv `find /var/cache/zypp/packages/ | grep aaa_base | head -n1` /tmp/aaa_base.rpm';
-
-    # List contents of an RPM package
-    assert_script_run 'rpm -qlp /tmp/aaa_base.rpm';
-
-    # List all packages that require a given package
-    assert_script_run 'rpm -q --whatrequires aaa_base';
-
-    # Dump basic file information of every file in a package
-    assert_script_run 'rpm -q --dump aaa_base';
-
-    # List requirements of an RPM package
-    assert_script_run 'rpm -qp --requires /tmp/aaa_base.rpm';
-
-    # Prepare test rpm file of missing package
-    assert_script_run('rpm -e sysstat') if (script_run("rpm -q sysstat") == 0);
-    zypper_call 'in -fy --download-only sysstat';
-    assert_script_run 'mv `find /var/cache/zypp/packages/ | grep sysstat | head -n1` /tmp/sysstat.rpm';
-
-    # Install prerequizities of sysstat package
-    zypper_call 'in procmail';
-
-    # Check if installation of a package will go through, do not actually install
-    assert_script_run 'rpm -ivh --test /tmp/sysstat.rpm';
-
-    # Install an RPM package
-    assert_script_run 'rpm -ivh /tmp/sysstat.rpm';
-
-    # Uninstall an already installed package
-    assert_script_run 'rpm -evh sysstat';
+    foreach my $pkg (@test_pkgs) {
+        record_info($pkg, script_output qq[ rpm -qpi $pkg]);
+        # Verify that the package has not been corrupted
+        assert_script_run("rpm -Kv $pkg",
+            fail_message => "Signature or digest does not match!\nPackage might be corrupted!");
+        # List contents of an RPM package
+        assert_script_run("rpm -qlp $pkg");
+        # List requirements of an RPM package
+        assert_script_run("rpm -qp --requires $pkg");
+        # Check if installation of a package will go through, then try real rpm deployment
+        unless (script_run("rpm --test -ivh $pkg")) {
+            # Install an RPM package
+            assert_script_run("rpm -ivh $pkg");
+        }
+        # Get rid of suffix, package has been already installed
+        my $installed_pkg = basename($pkg, '.rpm');
+        # Simple query for a package
+        assert_script_run("rpm -q $installed_pkg");
+        # List files in a package
+        assert_script_run("rpm -ql $installed_pkg");
+        # Get detailed information for a package
+        assert_script_run("rpm -qi $installed_pkg");
+        # Read changelog of a package
+        assert_script_run("rpm -q --changelog $installed_pkg");
+        # List what a package provides
+        assert_script_run("rpm -q --provides $installed_pkg");
+        # List all packages that require a given package
+        # Custom openqa_rpm_test package does not require any deps
+        if ((script_run("rpm -q --whatrequires $installed_pkg")) and ($pkg =~ /aaa_base/)) {
```

Comment:
It might be better outside loop with specific check:
```perl
   assert_script_run "rpm -q --whatrequires aaa_base"
   assert_script_run "rpm -q --whatrequires openqa_rpm_test | grep 'no package requires'"
```